결과를 CSV 파일로 출력

airotod opened the pull request:
### TODO
- [x] 결과를 CSV 파일로 출력 @airotod

airotod committed:
Update Config

- add path for saving csv

airotod committed:
Update BaseDataLoader

- add write sentence into csv file

airotod committed:
Update StandardEvaluator

- add write metric into csv file

airotod committed:
Update KGSFSystem

- add write test result
- add write prediction and response for conv into csv file

airotod committed:
Merge branch 'add-print-every-batch' into add-print-every-batch-csv

krta2 commented on crslab/evaluator/standard.py:
```diff
@@ -87,10 +87,15 @@ def gen_evaluate(self, hyp, refs):
             self.gen_metrics.add(k, AverageMetric(len(v) / self.dist_cnt))
         reports = [self.rec_metrics.report(), self.gen_metrics.report(), self.optim_metrics.report()]
+        if file:
+            for each_metric, value in self.rec_metrics.report().items():
+                file.write(f'{str(value.value())}\t')
+            for each_metric, value in self.gen_metrics.report().items():
+                file.write(f'{str(value.value())}\t')
```
@airotod 여기가 rec 일 때는 위에 for 문만 돌고 conv 일때는 아래 for 문만 도는거죠?

airotod replied:
rec일 때는 rec_metrics에 항목이 추가되고 conv일 때는 conv_metrics에 항목이 추가돼서 그렇게 돌아갑니다!

krta2 commented on crslab/system/kgsf.py:
```diff
@@ -186,12 +193,18 @@ def test_recommendation(self):
         self.init_optim(self.rec_optim_opt, self.model.parameters())
 
         logger.info('[Recommendation Test]')
+        # f = None
```
이 주석은 뭘까요?

airotod replied:
파일 출력할 때랑 아닐 때 테스트할 때 쓴 부분인데 필요 없는 부분이라 삭제할게요.

krta2 commented on crslab/system/kgsf.py:
```diff
@@ -206,12 +219,18 @@ def test_conversation(self):
         self.init_optim(self.conv_optim_opt, self.model.parameters())
 
         logger.info('[Conversation Test]')
+        # f = None
```
여기도요!

krta2 commented on crslab/system/kgsf.py:
```diff
@@ -55,6 +56,8 @@ def __init__(self, opt, train_dataloader, valid_dataloader, test_dataloader, voc
         self.rec_batch_size = self.rec_optim_opt['batch_size']
         self.conv_batch_size = self.conv_optim_opt['batch_size']
 
+        self.cpath = os.path.join(CSV_PATH, "kgsf", opt['dataset'])
```
```suggestion
        self.csv_path = os.path.join(CSV_PATH, "kgsf", opt['dataset'])
```
`csv_path` 가 분명해서 더 좋지 않을까요?

krta2 commented on crslab/config/__init__.py:
```diff
@@ -30,3 +30,4 @@
 MODEL_PATH = os.path.join(DATA_PATH, 'model')
 PRETRAIN_PATH = os.path.join(MODEL_PATH, 'pretrain')
 EMBEDDING_PATH = os.path.join(DATA_PATH, 'embedding')
+CSV_PATH = os.path.join(ROOT_PATH, 'result')
```
```suggestion
CSV_PATH = os.path.join(ROOT_PATH, 'test_result')
```
test_result 어떠신가요?!

krta2 commented on crslab/system/kgsf.py:
```diff
@@ -75,7 +78,11 @@ def conv_evaluate(self, prediction, response):
                 logger.info(f'\n   prediction: {p_str}\n   response: {r_str}')
```
여기도 get_data 에서처럼 file 이 없을때만 logger 로 출력하는거 어떠신가요..! 안그러면 터미널에 출력이 많아서 진행 정도 확인하기가 어려울거 같아서요..!

krta2 reviewed (commented):
@airotod 깔끔하네요! 수고 많으셨습니다!

krta2 commented on crslab/data/dataloader/base.py:
```diff
@@ -66,12 +66,20 @@ def get_data(self, batch_fn, batch_size, shuffle=True, process_fn=None):
+                        file.write('"')
+                        for sentence_in_index in conv_dict['context_tokens']:
+                            sentence = " ".join([self.vocab['ind2tok'][index] for index in sentence_in_index])
+                            logger.info(sentence)
```
앗 여기도 logger 가 있었구나
file 에 출력할 때는 logger 로 sentence 출력 안 해도 되지 않을까요..?

airotod committed:
Update Config

- update path for saving csv

airotod committed:
Update BaseDataLoader

- not log when writing csv file

airotod committed:
Update StandardEvaluator

- not log when writing csv file

airotod committed:
Update KGSFSystem

- update csv path variable name
- not log when writing csv file
- remove unnecessary comment

krta2 reviewed (approved):
@airotod 수고하셨습니다!
